core/src/main/java/org/infinispan/marshall/core/InternalClassIds.java

```diff
@@ -0,0 +1,21 @@
+package org.infinispan.marshall.core;
+
+/**
+ * Identifiers for marshalling class name itself; this comes handy when marshalling arrays
+ * and the array component type is interface (and therefore it does not have its AdvancedExternalizer).
+ *
+ * This identifiers don't clash with {@link org.infinispan.commons.marshall.Ids} and therefore it can use the same range.
+ */
+public interface InternalClassIds {
```

Comment:
`ClassNameIds` sounds fine. It is used for array component types, but it does not host array classes.

Right, it should not be public class. But interface allows me to omit the `static final`.